SYCL/Basic/device-selectors-exception.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_FILTER="" %t.out
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+int refuse_any_device_f(const device &d) { return -1; }
+
+int main() {
+
+  // Disable all available devices
+  setenv("SYCL_DEVICE_FILTER", "", 0);
+
+  // Check exception message for custom device selector
+  try {
+    queue custom_queue(refuse_any_device_f);
+  } catch (exception &E) {
+    assert(std::string(E.what()).find("info::device_type::") ==
```

Comment:
(find(something) == std::string::npos) means that "something" was not found, but the check should verify that it was found, right?